test/LayoutSpec.js

```diff
@@ -1,156 +1,104 @@
+        // when
+        const output = await layoutProcess(xml);
 
-    const originalContents = generatedDiagrams.map(
-      diagram => fs.readFileSync(__dirname + '/fixtures/' + diagram, 'utf8')
-    );
+        if (fs.existsSync(path.join(snapshotsDirectory, fileName))) {
+          const snapshot = fs.readFileSync(path.join(snapshotsDirectory, fileName), 'utf8');
 
-    const generatedContents = generatedDiagrams.map(
-      diagram => fs.readFileSync(outputDirectory + diagram, 'utf8')
-    );
+          // then
+          assert.strictEqual(output, snapshot, `Snapshot does not match output for ${ fileName }`);
+        }
 
-    const generated = generatedContents.map((contents, idx) => {
+        fs.writeFileSync(path.join(outputDirectory, fileName), output, 'utf8');
 
-      return {
-        diagram: generatedDiagrams[idx],
-        generatedContents: generatedContents[idx],
-        originalContents: originalContents[idx]
-      };
+        if (UPDATE_SNAPSHOTS) {
+          fs.writeFileSync(path.join(snapshotsDirectory, fileName), output, 'utf8');
+        }
+      });
     });
 
-    const config = JSON.stringify(generated);
-
-    const html = `
-<html>
-  <head>
-    <title>bpmn-auto-layouter - visual tests</title>
-    <style>
-      .title {
-        font-family: monospace;
-        margin-bottom: 10px;
-        margin-top: 10px;
-        font-weight: bold;
-      }
-
-      .container {
-        height: 400px;
-        width: 48%;
-        display: inline-block;
-        position: relative;
-        border: solid 1px #CCC;
-      }
 
-      .container + .container {
-        margin-left: 5px;
-      }
-
-      .parent {
-        margin-bottom: 30px;
-      }
-    </style>
-    <link rel="stylesheet" href="https://unpkg.com/bpmn-js/dist/assets/diagram-js.css">
-    <link rel="stylesheet" href="https://unpkg.com/bpmn-js/dist/assets/bpmn-js.css">
-  </head>
-  <body>
-    <script src="https://unpkg.com/bpmn-js/dist/bpmn-viewer.production.min.js"></script>
-    <script>
-      const config = ${config};
-
-      for (const { diagram, generatedContents, originalContents } of config) {
-
-        const parent = document.createElement('div');
-        parent.className = 'parent';
-
-        const title = document.createElement('div');
-        title.className = 'title';
-        title.textContent = 'test/generated/' + diagram;
-
-        const containerParent = document.createElement('div');
-        containerParent.className = 'container-parent';
-
-        const containerBefore = document.createElement('div');
-        containerBefore.className = 'container before';
-
-        const containerAfter = document.createElement('div');
-        containerAfter.className = 'container after';
-
-        parent.appendChild(title);
-        parent.appendChild(containerParent);
-
-        containerParent.appendChild(containerBefore);
-        containerParent.appendChild(containerAfter);
+  after(() => {
```

Comment:
Quite unexpected to me to see the teardown code on the bottom. Usual practice is to have tests setup on the top, and place test cases later:

before
after
it